Tiger does not have the _DARWIN_C_SOURCE override to enable chunks of headers
when _POSIX_C_SOURCE is defined, so we have to specifically remove the
_POSIX_C_SOURCE from these files for Tiger builds to succeed.

This patch could not be sent upstream as is... and probably would not be accepted
if it were sent upstream given the age of Tiger.

kencu@macports.org
--- src/sleep.c.orig	2020-04-06 15:21:04.000000000 -0500
+++ src/sleep.c	2020-05-15 07:56:47.000000000 -0500
@@ -33,7 +33,9 @@
 */
 
 #include "config.h"
+#if (__APPLE__ && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 1050)
 #define _POSIX_C_SOURCE 200112L
+#endif
 
 #include <errno.h>
 #include <time.h>
--- src/tcp.c.orig	2020-04-06 15:21:04.000000000 -0500
+++ src/tcp.c	2020-05-15 07:55:17.000000000 -0500
@@ -40,7 +40,9 @@
 
 /* TCP control file */
 
+#if (__APPLE__ && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 1050)
 #define _POSIX_C_SOURCE 200112L
+#endif
 
 #include "config.h"
 #include <sys/types.h>
